internal/sdkprovider/service/vpc/sweep.go

```diff
@@ -0,0 +1,264 @@
+//go:build sweep
+
+package vpc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aiven/aiven-go-client/v2"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/aiven/terraform-provider-aiven/internal/sweep"
+)
+
+func init() {
+	ctx := context.Background()
+
+	client, err := sweep.SharedClient()
+	if err != nil {
+		panic(fmt.Sprintf("error getting client: %s", err))
+	}
+
+	allServices := []string{
+		"aiven_pg",
+		"aiven_cassandra",
+		"aiven_opensearch",
+		"aiven_grafana",
+		"aiven_influxdb",
+		"aiven_redis",
+		"aiven_mysql",
+		"aiven_kafka",
+		"aiven_kafka_connect",
+		"aiven_kafka_mirrormaker",
+		"aiven_m3db",
+		"aiven_m3aggregator",
+		"aiven_flink",
+		"aiven_clickhouse",
+	}
+
+	resource.AddTestSweepers("aiven_project_vpc", &resource.Sweeper{
+		Name: "aiven_project_vpc",
+		F:    sweepVPCs(ctx, client),
+		Dependencies: []string{
+			"aiven_project",
+		},
+	})
+
+	resource.AddTestSweepers("aiven_aws_vpc_peering_connection", &resource.Sweeper{
+		Name: "aiven_aws_vpc_peering_connection",
+		F:    sweepVPCPeeringCons(ctx, client),
+		Dependencies: []string{
+			"aiven_project_vpc",
+		},
+	})
+
+	resource.AddTestSweepers("aiven_azure_vpc_peering_connection", &resource.Sweeper{
+		Name: "aiven_azure_vpc_peering_connection",
+		F:    sweepVPCPeeringCons(ctx, client),
+		Dependencies: []string{
+			"aiven_project_vpc",
+		},
+	})
+
+	resource.AddTestSweepers("aiven_gcp_vpc_peering_connection", &resource.Sweeper{
+		Name: "aiven_gcp_vpc_peering_connection",
+		F:    sweepVPCPeeringCons(ctx, client),
+		Dependencies: []string{
+			"aiven_project_vpc",
+		},
+	})
+
+	resource.AddTestSweepers("aiven_transit_gateway_vpc_attachment", &resource.Sweeper{
+		Name: "aiven_transit_gateway_vpc_attachment",
+		F:    sweepVPCPeeringCons(ctx, client),
+		Dependencies: []string{
+			"aiven_project_vpc",
+		},
+	})
+
+	resource.AddTestSweepers("aiven_aws_privatelink", &resource.Sweeper{
+		Name:         "aiven_aws_privatelink",
+		F:            sweepAWSPrivatelinks(ctx, client),
+		Dependencies: allServices,
+	})
+
+	resource.AddTestSweepers("aiven_azure_privatelink", &resource.Sweeper{
+		Name:         "aiven_azure_privatelink",
+		F:            sweepAzurePrivatelinks(ctx, client),
+		Dependencies: allServices,
+	})
+
+	resource.AddTestSweepers("aiven_gcp_privatelink", &resource.Sweeper{
+		Name:         "aiven_gcp_privatelink",
+		F:            sweepGCPPrivatelinks(ctx, client),
+		Dependencies: allServices,
+	})
+}
+
+func sweepVPCs(ctx context.Context, client *aiven.Client) func(string) error {
+	return func(region string) error {
+		projectName := os.Getenv("AIVEN_PROJECT_NAME")
+		vpcs, err := client.VPCs.List(ctx, projectName)
+		if err != nil {
+			return fmt.Errorf("error retrieving a list of vpcs for a project : %s", err)
+		}
+
+		for _, vpc := range vpcs {
+
+			err := client.VPCs.Delete(ctx, projectName, vpc.ProjectVPCID)
+			if err != nil && !aiven.IsNotFound(err) {
```

Comment:
let's do this: https://github.com/aiven/terraform-provider-aiven/pull/1525#discussion_r1448996423